zerver/actions/realm_settings.py

```diff
@@ -104,6 +105,43 @@ def do_set_realm_property(
         update_users_in_full_members_system_group(realm, acting_user=acting_user)
 
 
+@transaction.atomic(durable=True)
+def do_set_realm_permission_group_settings(
+    realm: Realm, name: str, user_group: UserGroup, *, acting_user: Optional[UserProfile]
+) -> None:
+    """Takes in a realm object, the name of an attribute to update, the
+    user_group to update and and the user who initiated the update.
+    """
+
+    old_user_group_id = getattr(realm, name)
+    setattr(realm, name, user_group.id)
+    realm.save(update_fields=[name])
+
+    event = dict(
+        type="realm",
+        op="update",
+        property=name,
+        value=user_group.id,
+    )
+
+    transaction.on_commit(lambda: send_event(realm, event, active_user_ids(realm.id)))
```

Comment:
This is an old pattern; switch this to `send_event_on_commit`. @HemantUmre12 @sahil839 FYI too since you may have some PRs that need this tweak too.